Sources/Network/NetworkData.swift

```diff
@@ -64,11 +67,21 @@ open class NetworkDataProcedure<Session: URLSessionTaskFactory>: Procedure, Inpu
         }
 
         stateLock.withCriticalScope {
-            guard !isCancelled else { return }
+            guard !isCancelled else {
+                finish()
+                return
+            }
             task = session.dataTask(with: request) { [weak self] data, response, error in
                 guard let strongSelf = self else { return }
 
                 if let error = error {
+                    let nsError = error as NSError
```

Comment:
FYI, the appropriate way to perform this check is to cast to `URLError` and check for the `.cancelled` case.

```swift
if let error = error {
    if strongSelf.isCancelled, let error = error as? URLError, error == .cancelled {
        strongSelf.finish()
        return
    }

    strongSelf.finish(withResult: .failure(error))
    return
}
```